### PR TITLE
Now pistons move in adjacent trusted regions. Configurable in config.yml

### PR DESCRIPTION
**Pull request by DevSlashRichie**
Basically what the titles say. 

I added this because some players wanted to coop but they weren't able to achieve this.
This can be modified (to the old behavior) in the configuration file. Default true.

### CODE REVIEW

**Comment by Jikoo** [comment]
The old behavior is reflected in the new settings, you probably were used to the equivalent of `EVERYWHERE`.

File: src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java [hunk 1]
```diff
@@ -597,6 +598,7 @@ else if (world.getEnvironment() == Environment.NORMAL)
         this.config_limitTreeGrowth = config.getBoolean("GriefPrevention.LimitTreeGrowth", false);
         this.config_pistonExplosionSound = config.getBoolean("GriefPrevention.PistonExplosionSound", true);
         this.config_pistonMovement = PistonMode.of(config.getString("GriefPrevention.PistonMovement", "CLAIMS_ONLY"));
+        this.config_pistonAllowIntersectionTrustedClaims = config.getBoolean("GriefPrevention.Piston", true);
```
Comment (Jikoo) [inline src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java]: Assuming this is a missed string - `GriefPrevention.Piston` is terribly undescriptive.

**Comment by Jikoo** [comment]
Overall not a fan of the handling. At a glance it looks like it undoes a lot of the performance improvements of the various modes. You're adding multiple iterations over both sets of blocks, that's a huge hit.

File: src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java (hunk 1 above)
Comment (DevSlashRichie) [reply src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java]: Oh, yea it is.

**Comment by DevSlashRichie** [comment]
> 
> 
> The old behavior is reflected in the new settings, you probably were used to the equivalent of `EVERYWHERE`.

I meant old behavior, as before my commit (without allowing pistons on adjacent trusted claims).

**Comment by DevSlashRichie** [comment]
@Jikoo What would be the best way to handle what I tried to achieve in your opinion?

**Comment by Jikoo** [comment]
Frankly I haven't looked it over hard, but the main thing with the piston stuff is that any operation will be called potentially thousands of times in quick succession, so microoptimizations are actually worth it.

The big thing that stands out at a glance is that you collect a stream and then iterate over it again - you're effectively doubling the iterations over the contents for no gain. It would be better to not filter using a stream at all, just use a for loop and continue.

**Pull request closed without merging**